packages/headless/src/controllers/product-listing/category-facet/headless-product-listing-category-facet.ts

```diff
@@ -68,9 +70,31 @@ export function buildCategoryFacet(
   const {dispatch} = engine;
   const getFacetId = () => coreController.state.facetId;
 
+  const createFacetSearch = () => {
+    const {facetSearch} = props.options;
+
+    return buildCategoryFacetSearch(engine, {
```

Comment:
I think it would be fine to make a core facet search and have a listing facet search. I don't really like passing a method to the initialization of a controller, I see controllers as something that abstracts away actions and the redux state.

Reply:
It duplicates code so much compared to the simplicity of what I have here :/

But I don't really mind in the end if you prefer splitting them off

Reply:
It's a tough one hmm. I know there a bit of work involved, but personally I would prefer if we followed the same pattern as with the other core controllers 🙏

Reply:
Is it a problem that leaks externally though ?

I must admit I'm reading this PR relatively rapidly, but it seems that no public symbol would ultimately be affected by this decision, right ? 

Like, if we had * one * public controller that had a very different signature than all others, I would be very much against it.

But, if it's only internal then... I'd be fine with passing a function as a prop, personally.

Reply:
I kinda agree, but @Spuffynism is already working on splitting the controllers, if that changes anything :man_shrugging:

Reply:
https://github.com/coveo/ui-kit/pull/1263/commits/66cfb98d4651c0edd274161d1f51c018a25cc128

:P